node/src/lib.rs

```diff
@@ -49,24 +35,33 @@ use signal_hook::{
     flag,
 };
 
-pub use components::{
-    block_proposer::Config as BlockProposerConfig,
-    consensus::Config as ConsensusConfig,
+pub(crate) use components::{
+    block_proposer::Config as BlockProposerConfig, consensus::Config as ConsensusConfig,
     contract_runtime::Config as ContractRuntimeConfig,
     deploy_acceptor::Config as DeployAcceptorConfig,
-    event_stream_server::Config as EventStreamServerConfig,
-    fetcher::Config as FetcherConfig,
-    gossiper::{Config as GossipConfig, Error as GossipError},
-    linear_chain_sync::Config as LinearChainSyncConfig,
-    rest_server::Config as RestServerConfig,
-    rpc_server::{rpcs, Config as RpcServerConfig},
-    small_network::{Config as SmallNetworkConfig, Error as SmallNetworkError},
-    storage::{Config as StorageConfig, Error as StorageError},
+    event_stream_server::Config as EventStreamServerConfig, fetcher::Config as FetcherConfig,
+    gossiper::Config as GossipConfig, linear_chain_sync::Config as LinearChainSyncConfig,
+    rest_server::Config as RestServerConfig, rpc_server::Config as RpcServerConfig,
+    small_network::Config as SmallNetworkConfig, storage::Config as StorageConfig,
 };
-pub use config_migration::{migrate_config, Error as ConfigMigrationError};
-pub use data_migration::{migrate_data, Error as DataMigrationError};
-pub use types::NodeRng;
-pub use utils::OS_PAGE_SIZE;
+pub(crate) use types::NodeRng;
+pub(crate) mod components;
+mod config_migration;
+mod data_migration;
+pub(crate) mod effect;
+pub(crate) mod logging;
+pub(crate) mod protocol;
+pub(crate) mod reactor;
+#[cfg(test)]
+pub(crate) mod testing;
+pub(crate) mod tls;
+pub(crate) mod utils;
+
+// Public modules
+pub use components::rpc_server::rpcs;
+pub mod cli;
+pub mod crypto;
+pub mod types;
```

Comment:
I've moved all of the `mod` declarations above the `use` declarations  in c2860ed92ab602884d3ac08b971ed8f05bc5ae71

I recognize it is important to have `mod` declaration before `use` declarations to avoid name collisions.